src/main/java/net/sf/jabref/importer/fileformat/MedlineImporter.java

```diff
@@ -84,50 +140,532 @@ public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
+        for (AbstractText text : abs.getAbstractText()) {
+            for (Serializable textContent : text.getContent()) {
+                if (textContent instanceof String) {
+                    abstractText.add((String) textContent);
+                }
+            }
+        }
+        fields.put("abstract", join(abstractText, " "));
+    }
+
+    private void addPagination(HashMap<String, String> fields, Pagination pagination) {
+        String startPage = "";
+        String endPage = "";
+        for (JAXBElement<String> element : pagination.getContent()) {
+            if ("MedlinePgn".equals(element.getName().getLocalPart())) {
+                putIfNotNull(fields, "pages", fixPageRange(element.getValue()));
+            } else if ("StartPage".equals(element.getName().getLocalPart())) {
+                //it could happen, that the article has only a start page
+                startPage = element.getValue() + endPage;
+                putIfNotNull(fields, "pages", startPage);
+            } else if ("EndPage".equals(element.getName().getLocalPart())) {
+                endPage = element.getValue();
+                //but it should not happen, that a endpage appears without startpage
+                fields.put("pages", fixPageRange(startPage + "-" + endPage));
+            }
+        }
+    }
+
+    private String extractYear(String medlineDate) {
+        //The year of the medlineDate should be the first 4 digits
+        return medlineDate.substring(0, 4);
+    }
+
+    private void handleAuthors(HashMap<String, String> fields, AuthorList authors) {
+        List<String> authorNames = new ArrayList<>();
+        for (Author author : authors.getAuthor()) {
+            if (author.getCollectiveName() != null) {
+                Text collectiveNames = author.getCollectiveName();
+                for (Serializable content : collectiveNames.getContent()) {
+                    if (content instanceof String) {
+                        authorNames.add((String) content);
+                    }
+                }
+            } else {
+                String authorName = author.getLastName();
+                if (author.getForeName() != null) {
+                    authorName += ", " + author.getForeName();
+                }
+                authorNames.add(authorName);
+            }
+        }
+        fields.put("author", join(authorNames, " and "));
+    }
+
+    private String join(List<String> list, String string) {
+        return Joiner.on(string).join(list);
+    }
+
+    private void addDateRevised(HashMap<String, String> fields, DateRevised dateRevised) {
+        if ((dateRevised.getDay() != null) && (dateRevised.getMonth() != null) && (dateRevised.getYear() != null)) {
+            fields.put("revised", dateRevised.getDay() + "/" + dateRevised.getMonth() + "/" + dateRevised.getYear());
+        }
+    }
+
+    private void putIfNotNull(HashMap<String, String> fields, String medlineKey, String value) {
```

Comment:
I would suggest renaming it, to make it more clear:_
putIfValueNotNull
Otherwise it could be confused with the jjava standard method putIfAbsent